internal/cloud/realm/realm.go

```diff
@@ -212,6 +254,142 @@ func isValidLocation(l Location) bool {
+	AWSProviderRegionAPSoutheast1 ProviderRegion = "aws-ap-southeast-1"
+	AWSProviderRegionAPSoutheast2 ProviderRegion = "aws-ap-southeast-2"
+	AWSProviderRegionAPSouth1     ProviderRegion = "aws-ap-south-1"
+	AWSProviderRegionUSEast2      ProviderRegion = "aws-us-east-2"
+	AWSProviderRegionEUWest2      ProviderRegion = "aws-eu-west-2"
+	AWSProviderRegionSAEast1      ProviderRegion = "aws-sa-east-1"
+
+	// Azure provider regions that we currently support
+	AzureProviderRegionEastUS2       ProviderRegion = "azure-eastus2"
+	AzureProviderRegionWestUS        ProviderRegion = "azure-westus"
+	AzureProviderRegionWestEurope    ProviderRegion = "azure-westeurope"
+	AzureProviderRegionEastAsia      ProviderRegion = "azure-eastasia"
+	AzureProviderRegionSouthEastAsia ProviderRegion = "azure-southeastasia"
+
+	// GCP provider regions that we currently support
+	GCPProviderRegionUSCentral1  ProviderRegion = "gcp-us-central1"
+	GCPProviderRegionUSWest1     ProviderRegion = "gcp-us-west1"
+	GCPProviderRegionEuropeWest1 ProviderRegion = "gcp-europe-west1"
+	GCPProviderRegionAsiaSouth1  ProviderRegion = "gcp-asia-south1"
+)
+
+// ProviderRegionToLocation provides a quick lookup of a provider and region combination to
+// location values. This is necessary as the backing clusters no longer match our current
+// operating regions. In order to do backing cluster storage, we must leverage a lookup here.
+var ProviderRegionToLocation = map[ProviderRegion]Location{
+	ProviderRegionEmpty:              LocationVirginia,
+	AWSProviderRegionUSEast1:         LocationVirginia,
+	AWSProviderRegionUSWest2:         LocationOregon,
+	AWSProviderRegionEUCentral1:      LocationFrankfurt,
+	AWSProviderRegionEUWest1:         LocationIreland,
+	AWSProviderRegionAPSoutheast1:    LocationSingapore,
+	AWSProviderRegionAPSoutheast2:    LocationSydney,
+	AWSProviderRegionAPSouth1:        LocationMumbai,
+	AWSProviderRegionUSEast2:         LocationVirginia,
+	AWSProviderRegionEUWest2:         LocationIreland,
+	AWSProviderRegionSAEast1:         LocationSaoPaolo,
+	AzureProviderRegionEastUS2:       LocationVirginia,
+	AzureProviderRegionWestUS:        LocationOregon,
+	AzureProviderRegionWestEurope:    LocationFrankfurt,
+	AzureProviderRegionEastAsia:      LocationMumbai,
+	AzureProviderRegionSouthEastAsia: LocationSingapore,
+	GCPProviderRegionUSCentral1:      LocationVirginia,
+	GCPProviderRegionUSWest1:         LocationOregon,
+	GCPProviderRegionEuropeWest1:     LocationFrankfurt,
+	GCPProviderRegionAsiaSouth1:      LocationMumbai,
+}
+
+// ProviderRegion is the Realm app provider region
+type ProviderRegion string
+
+// String returns the ProviderRegion display
+func (p ProviderRegion) String() string { return string(p) }
+
+func (p ProviderRegion) Label() string {
+	return p.String()[strings.IndexByte(p.String(), '-')+1:]
+}
+
+// Type returns the ProviderRegion type
+func (p ProviderRegion) Type() string { return flags.TypeString }
```

Comment:
[nit] we should try to avoid having a mix of pointer and non pointer receiver methods.

Reply:
I was thinking it's fine here in the same style of the other receivers because these aren't updated the value. But open to discuss

Reply:
I'll take the "blame" for the intent here, it was definitely my strategy (when it came to these "custom flag types") to only have the receivers be pointers when they _needed_ to, rather than worry about method set consistency nor the cost of a value-copy

basically, letting the last point of your linked article trump the rest of reasoning:

> For types such as basic types, slices, and small structs, a value receiver is very cheap so unless the semantics of the method requires a pointer, a value receiver is efficient and clear.

open to alternative approaches here as truthfully I don't think the code impact is significant one way or another.  these are likely more meaningful conversations around very complex structs and/or ones with fairly large method sets

Reply:
It's fine the way it is. I spoke with Kush offline about the pointers and non pointer methods breaking if we made them consistent. I would say that last segment is about using pointer receivers in the case of complex structures regardless of "set/assignment" methods rather than mixing inconsistencies because the type is simple.